rest/src/main/java/fi/otavanopisto/pyramus/rest/StudentRESTService.java

```diff
@@ -1914,43 +1915,67 @@ public Response createCourseAssessment(@PathParam("STUDENTID") Long studentId, @
     
     // #1198: Create a worklist entry for this assessment, if applicable
     
-    WorklistItemTemplate template = worklistController.getTemplateForCourseAssessment(assessmentController.isRaisedGrade(courseAssessment));
+    boolean isRaisedGrade = assessmentController.isRaisedGrade(courseAssessment);
+    WorklistItemTemplate template = worklistController.getTemplateForCourseAssessment();
     if (template != null) {
       
-      // Student display name
-      
-      StringBuilder sb = new StringBuilder();
-      sb.append(student.getFirstName());
-      if (!StringUtils.isEmpty(student.getNickname())) {
-        sb.append(String.format(" \"%s\"", student.getNickname()));
-      }
-      if (!StringUtils.isEmpty(student.getLastName())) {
-        sb.append(String.format(" %s", student.getLastName()));
-      }
-      if (student.getStudyProgramme() != null) {
-        sb.append(String.format(" (%s)", student.getStudyProgramme().getName()));
-      }
-      String studentDisplayName = sb.toString();
-      
-      // Course display name
-      
-      sb = new StringBuilder();
-      sb.append(course.getName());
-      if (!StringUtils.isEmpty(course.getNameExtension())) {
-        sb.append(String.format(" (%s)", course.getNameExtension()));
-      }
-      String courseDisplayName = sb.toString();
+      // #1228: Use billing settings together with the template to create a proper billing row
+
+      CourseBillingRestModel courseBillingRestModel = worklistController.getCourseBillingRestModel();
+      if (courseBillingRestModel != null) {
+        
+        // Determine billing number from student's study programme
+        // (high school if applicable, elementary as fallback)
+        
+        boolean isHighSchoolStudent = StringUtils.equalsIgnoreCase("lukio",
+            student.getStudyProgramme().getCategory().getEducationType().getCode());
```

Comment:
Likely needs a null check as category or education type might be missing for certain study programmes